### PR TITLE
Feature/dynamic backend objects

### PR DESCRIPTION
**Pull request by mmontero**
Use r2inference meta to create the GStreamer's backend objects

### CODE REVIEW

File: gst-libs/gst/r2inference/gstbasebackendsubclass.h [hunk 1]
```diff
@@ -19,23 +19,25 @@
...
-void gst_backend_install_properties (GstBackendClass * klass,
+void gst_base_backend_install_properties (GstBaseBackendClass * klass,
                                 r2i::FrameworkCode code);
-gboolean gst_backend_set_framework_code (GstBackend * backend,
+gboolean gst_base_backend_set_framework_code (GstBaseBackend * backend,
                                          r2i::FrameworkCode code);
 
+gboolean gst_inference_backend_register (const gchar* type_name, r2i::FrameworkCode code);
+
 G_END_DECLS
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/r2inference/gstbasebackendsubclass.h:42]: What if we collapse this with the gstbasebackend.h? Not sure why we have those functions separated

File: gst-libs/gst/r2inference/gstbasebackend.cc [hunk 2]
```diff
@@ -102,42 +102,42 @@ struct _GstBackendPrivate {
 
 };
 
-G_DEFINE_TYPE_WITH_CODE (GstBackend, gst_backend, G_TYPE_OBJECT,
-                         GST_DEBUG_CATEGORY_INIT (gst_backend_debug_category, "backend", 0,
-                             "debug category for backend parameters"); G_ADD_PRIVATE (GstBackend));
+G_DEFINE_TYPE_WITH_CODE (GstBaseBackend, gst_base_backend, G_TYPE_OBJECT,
+                         GST_DEBUG_CATEGORY_INIT (gst_base_backend_debug_category, "backend", 0,
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstbasebackend.cc:106]: "backend" -> "basebackend"

File: gst-libs/gst/r2inference/gstinferencebackend.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+struct _GstInferenceBackend
+{
+  GstBaseBackend parent;
+};
+
+typedef struct _GstInferenceBackendClass GstInferenceBackendClass;
+struct _GstInferenceBackendClass
+{
+  GstBaseBackendClass parent_class;
+    r2i::FrameworkCode code;
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstinferencebackend.cc:38]: Use the no-indent trick here to fix the indentation.

File: gst-libs/gst/r2inference/Makefile.am [hunk 4]
```diff
@@ -44,13 +40,8 @@ gstinferenceinclude_HEADERS=		\
...
 	gstchildinspector.h		\
 	gstinferencebackends.h		\
-	gstbackend.h			\
-	gstbackendsubclass.h		\
-	gstedgetpu.h                    \
-	gstncsdk.h			\
-	gsttensorflow.h			\
-	gsttflite.h			\
-	gsttensorrt.h			\
+	gstbasebackend.h			\
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/Makefile.am:43]: Match backslashes

**Pull request merged**